### DISCUSSION TITLE
More Columns on Dashboard

### DISCUSSION THREAD

**Discussion by etmartinkazoo** [Thinkmill/keystatic#375] [Q&A]
In the below screenshot, the list of packages are present. Some packages have the same name and are only differentiated by their "tag" relationship. Some are wedding, some are corporate, and some are social. Is there a way to create a column called "Tag" that would display the name of the tag selected in the relationship field in addition to the "Name" column?

Thanks all!

![CleanShot 2023-06-27 at 13 50 45](https://github.com/Thinkmill/keystatic/assets/13721164/203671ec-91ec-405d-8742-2c5040d9a86d)

**Comment by JedWatson** [comment]
Hey @etmartinkazoo, thanks for the feedback, this is something we've been thinking about adding.

For background, the concern here is performance — right now, Keystatic can render that screen by just querying the list of files in the collection directory, which scales pretty well, but adding the ability to have columns from the frontmatter / json / etc would potentially mean actually reading a large number of files before the screen can be rendered.

For small collections that's obviously not going to be an issue but we're wary of adding foot guns.

We're considering adding some indexing (possibly optional) to the system, which would make this possible, or at least making the less-performant option available for now and show a warning if it's used with large collections.

Long story short, it's not impossible, just not necessarily trivial. We're going to come out with a solve for it before too long, and it's helpful to know you've got the use case so we can prioritise it.

Related, we've also discussed adding sorting and filtering on fields other than the slug which also has implications on how we build support for showing fields in the table, so shout out if that's something that would be super useful too.

**Reply by etmartinkazoo** [reply]
@JedWatson - Thanks for the feedback and I appreciate all the work everyone is doing. Nothing done well is simple so I'm happy to be along for the ride.

I don't have any hard data on this but my gut reaction is that most sites are not big enough - I'm thinking the thousands and thousands of small to medium sized businesses - to impact performance prohibitively but I can definitely see how this would be an issue at scale. I love the idea of indexing.
